slu/slu/dev/prompt_setup.py

```diff
@@ -0,0 +1,269 @@
+"""
+Generate config for prompts using nls-keys downloaded from studio
+Usage:
+    prompt_setup.py
+Options:
+    -h --help   Show this screen.
+    --version   Show version.
+"""
+
+import argparse
+import os
+import re
+import string
+
+import pandas as pd
+import yaml
+from tqdm import tqdm
+
+from slu import constants as const
+from slu.utils import logger
+from slu.utils.config import Config, YAMLLocalConfig
+
+
+def preprocess_prompt(
+    prompt: str,
+    remove_var: bool = True,
+    fill_token: str = const.PROMPT_NOISE_FILLER_TOKEN,
+) -> str:
+
+    """
+    REGEX1: Detect special characters
+    REGEX2: Detect noisy digits, alphanumberic characters
+    REGEX3: Detect consequtive black spaces
+    REGEX4: Detect variables defined inside prompts. Eg: {{.variable}}
+    """
+
+    REGEX1 = re.compile("\[.*?\]")
+    REGEX2 = re.compile("<.*?>|&([a-z0-9]+|#[0-9]{1,6}|#x[0-9a-f]{1,6});")
+    REGEX3 = re.compile("/\W/g")
+    REGEX4 = re.compile(r"{{(.*?)}}")
+
+    if not isinstance(prompt, str):
+        return prompt
+
+    prompt = prompt.lower()
+    prompt = re.sub(REGEX1, "", prompt)
+    prompt = re.sub(REGEX2, "", prompt)
+    prompt = re.sub(REGEX3, "", prompt)
+    prompt = re.sub(" +", " ", prompt)
+    prompt = re.sub("_", " ", prompt)
+    prompt = prompt.strip()
+    if remove_var:
+        prompt = re.sub(REGEX4, fill_token, prompt)
+    prompt = prompt.translate(
+        str.maketrans("", "", string.punctuation.replace("<", "").replace(">", ""))
+    )
+
+    return prompt
+
+
+def _valid_string(string: str) -> bool:
+    if isinstance(string, str):
+        if all(
+            [
+                len(string) > 0,
+                string != "nan",
+                string != ".nan",
+                string != "",
+                string != " ",
+                "Unnamed" not in string,
+            ]
+        ):
+            return True
+    return False
+
+
+def _nls_to_state(string: str, delimiter: str = "_") -> str:
+    """
+    Convert NLS-Key to State.
+    NLS-Keys are nothing but extentions of State names.
+    Eg: state "A" will have NLS-Keys names A_1, A_2, and so on.
+    This code simply removes the suffix _1, _2, etc from a NLS-Key to get the original State name.
+    """
+    if not _valid_string(string):
+        return None
+    string = string.split(delimiter)
+    if len(string) > 1:
+        string = str(delimiter).join(_ for _ in string[: len(string) - 1])
+    if isinstance(string, list) and len(string) > 0:
```

Comment:
`string.split` will never return an empty list, will it?

Reply:
Right, changed the condition.